patches/0076-qsv-use-a-new-method-to-create-mfx-session-when-usin.patch

```diff
@@ -1007,6 +1013,44 @@ index a8bc8c8143..d3c061b739 100644
      return ret;
  }
  
+@@ -1523,6 +1746,13 @@ static int qsv_device_create(AVHWDeviceContext *ctx, const char *device,
+         }
+     } else if (CONFIG_VAAPI) {
+         child_device_type = AV_HWDEVICE_TYPE_VAAPI;
++#if QSV_ONEVPL
++    } else if (CONFIG_D3D11VA) {  // Use D3D11 by default if d3d11va is enabled
++
++        child_device_type = AV_HWDEVICE_TYPE_D3D11VA;
++    } else if (CONFIG_DXVA2) {
++        child_device_type = AV_HWDEVICE_TYPE_DXVA2;
++#else
+     } else if (CONFIG_DXVA2) {
+         av_log(NULL, AV_LOG_WARNING,
+                 "WARNING: defaulting child_device_type to AV_HWDEVICE_TYPE_DXVA2 for compatibility "
+@@ -1531,6 +1761,7 @@ static int qsv_device_create(AVHWDeviceContext *ctx, const char *device,
+         child_device_type = AV_HWDEVICE_TYPE_DXVA2;
+     } else if (CONFIG_D3D11VA) {
+         child_device_type = AV_HWDEVICE_TYPE_D3D11VA;
++#endif
+     } else {
+         av_log(ctx, AV_LOG_ERROR, "No supported child device type is enabled\n");
+         return AVERROR(ENOSYS);
+@@ -1557,6 +1788,15 @@ static int qsv_device_create(AVHWDeviceContext *ctx, const char *device,
+ #endif
+ #if CONFIG_DXVA2
+     case AV_HWDEVICE_TYPE_DXVA2:
++#if QSV_ONEVPL
++        {
++            av_log(NULL, AV_LOG_WARNING,
++                   "WARNING: d3d11va is not available or child device type is "
++                   "set to dxva2 explicitly for oneVPL code. Note oneVPL "
++                   "runtimes support d3d11va only on Microsoft Windows. Please "
```

Comment:
dxva2 is supported only in compatibility mode and new oneVPL features may not be supported